src/threshold_sign.rs

```diff
@@ -179,7 +221,7 @@ impl<N: NodeIdT> ThresholdSign<N> {
             .netinfo
             .public_key_set()
             .public_key()
-            .verify_g2(&sig, self.msg_hash)
+            .verify_g2(&sig, self.msg_hash.unwrap())
```

Comment:
I'd like to avoid `unwrap` calls in production code: We could pass in the hash from `try_output`, which should check it anyway and return an empty step if `None`.